### DISCUSSION TITLE
Integrating datasets with multiple biological replicates and multiple conditions?

### DISCUSSION THREAD

**Discussion by ThomasBaardemans** [satijalab/seurat#5702] [Q&A]
Hello, 

I am currently analyzing a public single cell dataset. 
It contains datasets for multiple biological replicates across 2 conditions: patients and controls. 
To integrate the data I took all samples and integrated them in 1 go using the FindIntegrationAnchors() and IntegrateData() functions from Seurat as shown in "workflow 1" in the image below.

![integration_workflows](https://user-images.githubusercontent.com/49636966/157059266-57a86a31-b6a0-4f2c-9720-67a8a1d64a3e.jpg)

However when my resulting UMAP's and clusters did not look alot like the figures shown in the paper I looked at their code and found out they did the integration in 2 steps as shown in "workflow 2" in the image above. 

I have never come across someone doing it like this. But I am quite new to analyzing single cell datasets so now I am wondering:
Is it "correct" to do the integration in 2 steps or might this have adverse effects? 

Looking forward to hear your thoughts and thanks in advance!

Thomas

**Comment by ConDem94** [comment]
> Hello, 
> 
> 
> 
> I am currently analyzing a public single cell dataset. 
> 
> It contains datasets for multiple biological replicates across 2 conditions: patients and controls. 
> 
> To integrate the data I took all samples and integrated them in 1 go using the FindIntegrationAnchors() and IntegrateData() functions from Seurat as shown in "workflow 1" in the image below.
> 
> 
> 
> ![integration_workflows](https://user-images.githubusercontent.com/49636966/157059266-57a86a31-b6a0-4f2c-9720-67a8a1d64a3e.jpg)
> 
> 
> 
> However when my resulting UMAP's and clusters did not look alot like the figures shown in the paper I looked at their code and found out they did the integration in 2 steps as shown in "workflow 2" in the image above. 
> 
> 
> 
> I have never come across someone doing it like this. But I am quite new to analyzing single cell datasets so now I am wondering:
> 
> Is it "correct" to do the integration in 2 steps or might this have adverse effects? 
> 
> 
> 
> Looking forward to hear your thoughts and thanks in advance!
> 
> 
> 
> Thomas
> 
> 
> 
> 

Have you found a solution to this? I'm thinking of doing the same thing. Could you cite the paper?

**Comment by Dragonmasterx87** [comment]
Not a dev. Workflow 2 is reasonable. This is because the underlying assumption is that the drug perturbations will be common to all the Treated samples. Therefore it is reasonable to first integrate controls and then the treatments separately, generating 2x Seurat objects and then integrating the two together. You don't need to worry about it tho Seurat's got your back, it automates this process for you.

Essentially this can also be done using:
```
# Identify anchors and integrate multi-donor-multi Tx dataset
# Create a list containing 10 datasets. 5 are control (1, 3, 5, 7, 9) while 5 are treatments (2, 4, 6, 8, 10) 
# Here the list is called data.list
# Let me know if you need help understanding how to make a list by tagging me
data.list[c(1, 3, 5, 7, 9)] # check that you are correctly picking up control datasets for reference integration
data.anchors <- FindIntegrationAnchors(object.list = data.list, 
                                       reference = c(1,3,5,7,9), 
                                       reduction = "rpca", 
                                       dims = 1:30, 
                                       verbose = TRUE)
data.integrated <- IntegrateData(anchorset = data.anchors, dims = 1:30, verbose = TRUE)
```
You can then treat data.integrated as an integrated dataset where certain data are used as reference samples in this case samples 1, 3, 5, 7 and 9.

Enjoy!

🐲

**Reply by AFrolicOfFerns** [reply]
Hi @Dragonmasterx87 , were you using seurat v4 when you wrote that code? I'm trying to do this with seurat v5, but the integration workflow has changed. Thanks!

**Reply by Dragonmasterx87** [reply]
Hi @AFrolicOfFerns yes, this is using v4. A long time has passed since this conversation, but honestly it no longer matters, if you integrate across all thats fine. If you perform Harmony you will be able to remove batch effects during integration.

**Reply by AFrolicOfFerns** [reply]
Wow, a helpful response *and* within an hour?? Thank you so much, @Dragonmasterx87 ! I haven't tried harmony yet, but sounds like it'll be better for my needs.
